packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -367,6 +345,33 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
     });
   }
 
+  void _assertEntriesInsertedPosition(OverlayEntry above, OverlayEntry below,
+      {Iterable<OverlayEntry> newEntries}) {
```

Comment:
Maybe make this required and remove the if else below?

Reply:
we will also need to update the error text to be more generic if we merge this too check.